### ISSUE TITLE
Recent Emoji Still Saved in Incognito Mode

### ISSUE THREAD

**Issue by mister-sir** [openboard-team/openboard#521]
**Describe the bug**
When the keyboard is in incognito mode, any emoji typed are added to the "Recent Emoji" panel

**To Reproduce**
1. Check recent emoji
2. Swap to a different panel of emoji besides recent
3. Open the keyboard in any app / field that uses incognito mode
4. Type an emoji not already at the top of recent emoji
5. Swap back to the recent emoji panel
6. Observe the emoji at the top of the recent emoji list
7. This persists even after changing back to a non-incognito field / app (unsurprisingly)

**Expected behavior**
I would expect any emoji typed while in incognito mode would not be added to the recent emoji list.

**Smartphone (please complete the following information):**
 - Device: Motorola G7 Play
 - OS: Android 10

**Additional context**
Using version 1.4.4 of openboard.

**Comment by mister-sir** [comment]
I've made changes to the code to implement a fix, but I've having a heck of a time getting a development environment set up to build and test. I've got it forked on my account here on github, and would appreciate any feedback. I'm totally new to app development.

I also have two questions:
1. Should there be an incognito icon to the emoji keyboard while in incognito mode? Currently one is not displayed as it is in the normal keyboard mode.
2. Should it still save the most recent emoji category used (i.e. `mEmojiCategory.saveLastTypedCategoryPage();`) while in incognito? My code currently doesn't, as that seems more "incognito" to me...
